### ISSUE TITLE
SyntaxError Encountered during 3D Reconstruction Evaluation

### ISSUE THREAD

**Issue by KevinYu0910** [eriksandstroem/Point-SLAM#8]
Hello,

I encountered an issue while running the following command:
```bash
python run.py configs/Replica/room0.yaml"
```
Initially, everything worked successfully, but some errors occurred in the later stages during the evaluation of the reconstruction. Here is the relevant information and error traceback from the execution log:

⭐ INFO: Output, GT, generated, residual images, and meshes are stored in various subdirectories under output/Replica/room0/20231009_191508.
⭐ Meshing finished successfully and evaluation started.

However, an error interrupted the process:
```bash
Traceback (most recent call last):
  File "/data/data_2t/Point-SLAM/src/Mapper.py", line 923, in run
    result_dict = literal_eval(result_dict)
  File "/home/kevin/anaconda3/envs/point-slam/lib/python3.10/ast.py", line 62, in literal_eval
    node_or_string = parse(node_or_string.lstrip(" \t"), mode='eval')
  File "/home/kevin/anaconda3/envs/point-slam/lib/python3.10/ast.py", line 50, in parse
    return compile(source, filename, mode, flags,
  File "<unknown>", line 0
    
SyntaxError: invalid syntax

```

The evaluation failed with a SyntaxError.

I appreciate any help or suggestions on how to resolve this issue. Thank you in advance!

Best regards
Kevin

**Comment by eriksandstroem** [comment]
Hi Kevin,
It seems the mesh was successfully created. It is hard to say what failed. To debug this issue, you can run the function `get_mesh_tsdf_fusion.py`. This is what is run to perform the evaluation. You can check in the `mapper.py` how this is called. You need to feed some inputs to it.

**Comment by KevinYu0910** [comment]
Hi @eriksandstroem,
Thank you again for your guidance. I've attempted to troubleshoot and test ```mapper.py``` as per your suggestions and made some findings.

I discovered that both ```start_index``` and ```end_index``` variables are receiving a value of ```-1```, indicating that the target string is not found. This, in turn, results in ```result_dict``` being empty and subsequently leads to the later ```SyntaxError: invalid syntax```.

I followed the execution instructions based on your GitHub guidance. Could you please advise if there are any additional modifications or steps that I might need to undertake to successfully execute the code without encountering this issue?

Thank you in advance for your further assistance.

Best regards,
Kevin

**Comment by eriksandstroem** [comment]
Did you run the `get_mesh_tsdf_fusion.py` script? If so, was it working? If not, I would suspect that something went wrong with the mesh evaluation. If you have the log file, feel free to send me this at my email and I can check it.

**Comment by KevinYu0910** [comment]
Thank you for your quick response.

I have successfully run the get_mesh_tsdf_fusion.py script, and here is the output confirming its successful execution:
```bash===========================
Evaluating room0
===========================
Rigid Transform Applied to Reconstructed Mesh:  [[ 9.99997392e-01  5.01624822e-04 -2.22813756e-03  1.43732111e-03]
 [-5.02544121e-04  9.99999789e-01 -4.12045359e-04  1.27095531e-03]
 [ 2.22793039e-03  4.13164022e-04  9.99997433e-01 -5.03373655e-04]
 [ 0.00000000e+00  0.00000000e+00  0.00000000e+00  1.00000000e+00]]
[Evaluate]
[compute distance from source to target]
[compute distance from target to source]

✨ Successfully evaluated 3D reconstruction.```
```
Regarding your note about ```cfg['meshing']['eval_rec']``` being set to True, I've verified that in mapper.py, we have ```self.eval_rec = cfg['meshing']['eval_rec']```, and within the ```config/point_slam.yaml ```file, ```eval_rec: True``` is indeed present, so it seems like it hasn't been changed to False.

Is the log file you mentioned generated by wandb?


Thank you again for your time and assistance.

**Comment by eriksandstroem** [comment]
No problem. Happy to assist. I was thinking about the stdout log (i.e. everything that is printed when running the code).

**Comment by KevinYu0910** [comment]
Hello,

My name is Kevin, and I have left a message in the issue section. Attached
are documents generated by wandb. The "debug.log" and "debug-internal.log"
are located under the wandb logs folder, while the "output.log" is in the
file folder. If there’s anything else you need, please let me know. Thank
you for your assistanc

Erik Sandström ***@***.***> 於 2023年10月11日 週三 下午9:42寫道：

**Comment by eriksandstroem** [comment]
Hi Kevin!
Hmm, strangely I don't see any attachments... :(

Can you send to ***@***.***?



Den ons 11 okt. 2023 kl 16:45 skrev KevinYu0910 ***@***.***>:

**Comment by KevinYu0910** [comment]
Did you not receive the email I sent? If not, could you please provide your email address? Alternatively, is there another way for me to send you the log file?"

**Comment by eriksandstroem** [comment]
> Did you not receive the email I sent? If not, could you please provide your email address? Alternatively, is there another way for me to send you the log file?"

Feel free to send to erik.sandstrm@gmail.com

**Comment by KevinYu0910** [comment]
Thank you for your ongoing assistance.

I have sent the log file to you via email as per your request. I am wondering if the aforementioned error is related to the ```evaluate_3d_reconstruction_lib``` that you provided on GitHub. I followed your steps :
```bash
git clone https://github.com/tfy14esa/evaluate_3d_reconstruction_lib.git
cd evaluate_3d_reconstruction_lib
pip install .```
```
The file path is set to ```/Point-SLAM/evaluate_3d_reconstruction_lib/```

I'm wondering if the issue might be related to the file path or perhaps a usage misstep on my part. Any insights or advice you could provide would be immensely helpful.

Thank you again for your time and assistance.

**Comment by eriksandstroem** [comment]
I checked your output log and indeed, the 3D evaluation fails since it outputs "Failed to evaluate 3D reconstruction". 

Can you try running the evaluation script on the meshes by running the python script here: https://github.com/eriksandstroem/Point-SLAM/blob/388d60922053c5ab4b8ae8f0f26d8d73e86d3136/src/tools/get_mesh_tsdf_fusion.py#L364

Run it directly in the terminal. Then we can debug easier.

**Comment by KevinYu0910** [comment]
Thank you for your valuable tool and advice! I directly ran ```eval_recon.py``` in the terminal as per your suggestion and encountered an error, which stated ```No such file or directory: 'latex'```. I went ahead and installed it, and although there were a few minor issues afterward, I was able to successfully execute it. Subsequently, I re-ran the ```run.py``` file, and it worked successfully as well.
Thank you again for your time and assistance.
